### ISSUE TITLE
[Customer]failing to specify custom kernel, initrd, kcmdline via boottarget

### ISSUE THREAD

**Issue by mbausch0** [xcat2/xcat-core#553]
xcat 2.11 (git commit 9ea36ca6163392bf9ab684830217f017193815be, built Mon Nov 30 05:43:11 EST 2015)
arch: ppc64 (IBM Power8)
OS: SLES for SAP Applications  11.4 (ppc64)

osimage provisioning using the osimage attribute "boottarget" to specify customized kernel, initrd (and kcmdline) fails to create yaboot.conf according to boottarget specifications.
Instead, boottarget specifications are completely ignored.

ismgp92:~ # lsdef -t osimage myimage
Object name: myimage
    boottarget=mytarget
    imagetype=linux
    osarch=ppc64
    osdistroname=sles11sp4sap-ppc64
    osname=Linux
    osvers=sles11sp4sap
    otherpkgdir=/install/post/otherpkgs/sles11sp4sap/ppc64
    pkgdir=/install/repos/sles11sp4sap/ppc64
    pkglist=/opt/xcat/share/xcat/install/sles/compute.pkglist
    profile=compute
    provmethod=install
    template=/opt/xcat/share/xcat/install/sles/compute.tmpl

ismgp92:~ # tabdump boottarget
#bprofile,kernel,initrd,kcmdline,comments,disable
"mytarget","/install/repos/sles11sp4sap/ppc64/1/suseboot/linux64","/install/repos/sles11sp4sap/ppc64/1/suseboot/initrd64",,,

ismgp92:~ # nodeset test880m osimage=myimage
test880m: install sles11sp4sap-ppc64-compute

I would expect this to create a yaboot.conf file which will use linux64 and
initrd64 as specified in boottarget mytarget.
Instead of that, the generated yaboot.conf will boot kernel inst64 (not
linux64) and no initrd at all; xcat completely ignores my boottarget
specifications:

ismgp92:~ # cat /tftpboot/yaboot.conf-MACaddr
#install sles11sp4sap-ppc64-compute
timeout=5
image=xcat/osimage/myimage/inst64
        label=xcat
        append="quiet
autoyast=http://<Server>:<Port>/install/autoinst/test880m
install=http://<Server>:<Port>/install/repos/sles11sp4sap/ppc64/1
netdevice=<MACaddr>  dhcptimeout=150"

For further details, please see the corresponding discussion on the xcat-user mailing list:
http://sourceforge.net/p/xcat/mailman/message/34714873/

**Comment by cxhong** [comment]
@mbausch0 Thanks to open this issue.
Can u change your osimage like this: 
````
ismgp92:~ # lsdef -t osimage myimage
Object name: myimage
boottarget=mytarget
imagetype=linux
osarch=ppc64
osdistroname=sles11sp4sap-ppc64
osname=Linux
osvers=boottarget
otherpkgdir=/install/post/otherpkgs/sles11sp4sap/ppc64
pkgdir=/install/repos/sles11sp4sap/ppc64
pkglist=/opt/xcat/share/xcat/install/sles/compute.pkglist
profile=mytarget
provmethod=install
template=/opt/xcat/share/xcat/install/sles/compute.tmpl
````

**Comment by mbausch0** [comment]
the results are very similar to what we had yesterday with nodeset netboot:
kernel and initrd made it into yaboot.conf, but no append="..." line and no autoyast profile:

ismgp92:~ # lsdef -t osimage myimage
Object name: myimage
    boottarget=mytarget
    imagetype=linux
    osarch=ppc64
    osdistroname=sles11sp4sap-ppc64
    osname=Linux
    osvers=boottarget
    otherpkgdir=/install/post/otherpkgs/sles11sp4sap/ppc64
    pkgdir=/install/repos/sles11sp4sap/ppc64
    pkglist=/opt/xcat/share/xcat/install/sles/compute.pkglist
    profile=mytarget
    provmethod=install
    template=/opt/xcat/share/xcat/install/sles/compute.tmpl

ismgp92:~ # tabdump nodetype
#node,os,arch,profile,provmethod,supportedarchs,nodetype,comments,disable
"lpar",,"ppc64",,,,,,
"test880m","boottarget","ppc64","mytarget",,,,,

ismgp92:~ # tabdump boottarget
#bprofile,kernel,initrd,kcmdline,comments,disable
"mytarget","linux64","initrd64",,,

ismgp92:~ # nodeset test880m osimage=myimage
Warning: test880m: yaboot might be invalid when provisioning myimage,valid options: "grub2,grub2-tftp,grub2-http".
For more details see the 'netboot' description in the output of "tabdump -d noderes".
test880m: install boottarget-ppc64-mytarget
Error: stop configuration because yaboot DOES NOT work for boottarget provision, please change noderes.netboot=grub2 instead.

ismgp92:~ # echo $?
1


ismgp92:~ # cat /tftpboot/yaboot.conf-MACaddr
#install boottarget-ppc64-mytarget
timeout=5
image=linux64
        label=xcat
        initrd=initrd64

ismgp92:~ # tabdump nodetype
#node,os,arch,profile,provmethod,supportedarchs,nodetype,comments,disable
"lpar",,"ppc64",,,,,,
"test880m","boottarget","ppc64","mytarget","myimage",,,,

ismgp92:~ # ls -l /install/autoinst/test880m
ls: cannot access /install/autoinst/test880m: No such file or directory

**Comment by cxhong** [comment]
That Error message
`````
Error: stop configuration because yaboot DOES NOT work for boottarget provision, please change noderes.netboot=grub2 instead.
`````
Caused routine returned and didn't update the yaboot.conf file with "append="  line.

**Comment by cxhong** [comment]
Hi, @mbausch0 :

I checked the code, you need to change the netboot=grub2 (I hit disk error if I used yaboot to provision sles11.4 ppc64 nodes).  and you also need to add kcmdline in the boottarget table.

````
# tabdump boottarget
#bprofile,kernel,initrd,kcmdline,comments,disable
"mytarget","/install/repos/linux64","/install/repos/initrd64","quiet sysrq=1 insmod=sym53c8xx insmod=ipr",,
````

then run nodeset command:
````
 # nodeset c910f02c06p13 osimage=myimage
c910f02c06p13: install boottarget-ppc64-mytarget
````

the custom kernel. initrd, and kcmdline will show up at /tftpboot/boot/grub2

````
 # cat grub.cfg-01-5e-cc-c7-8f-45-03
#install boottarget-ppc64-mytarget
set timeout=5
set default="xCAT OS Deployment"
menuentry "xCAT OS Deployment" {
    insmod http
    insmod tftp
    set root=tftp,10.2.6.11
    echo Loading Install kernel ...
    linux //install/repos/linux64 quiet sysrq=1 insmod=sym53c8xx insmod=ipr
    echo Loading initial ramdisk ...
    initrd //install/repos/initrd64
````

**Comment by mbausch0** [comment]
Hi, @cxhong 

thanks very much for your help.
I tried netboot=grub2
This exactly works, as you described it.

This way, two things are still missing for a successful node installation:
1. The autoyast file does not get created.
2. The following kernel arguments are missing in grub.cnf: autoyast=..., install=..., netdevice=<MACaddr>, dhcptimeout=150

I could add the missing arguments to the boottarget table with the exception of netdevice, because the MAC is different for every node.
And doing so, the autoyast file would still need to get created.

What needs to be done to also resolve these 2 items?

**Comment by zet809** [comment]
@xuweibj will you pls take a look at this issue?

**Comment by xuweibj** [comment]
https://sourceforge.net/p/xcat/wiki/BootTarget/
According to this doc, user must add bprofile, kernel, initrd and kcmdline to boottarget table, and prepare related files.

**Comment by mbausch0** [comment]
> https://sourceforge.net/p/xcat/wiki/BootTarget/

Yes, this is where the whole discussiion initially started:
https://sourceforge.net/p/xcat/mailman/message/34714873/


> According to this doc, user must add bprofile, kernel, initrd and kcmdline to boottarget table,
> and prepare related files.

Yes, this is exactly, what I did.
The only problem is: It doesn't work.
For what has already been suggested to solve this and how it worked out, please see this issue's history.
The question still remains: What needs to be done to make this work?

**Comment by cxhong** [comment]
Hi, @mbausch0 , Can u able to copy your definition here?

**Comment by daniceexi** [comment]
@mbausch0 

*  You are using boottarget method, which means xCAT does not know which OS the you want to install. Then xCAT cannnot figure out how to generate the configuration file like autoyast for you.

* You need to specify all the parameters by yourself through table boottarget.kcmdline. kcmdline supports two specific formats to simplify the input:

  * #NODEATTRIB:<tablename>:<columnname>
  * #TABLE:<tablename>:<nodename>:<columnname>

**Comment by whowutwut** [comment]
@xuweibj @tingtli  If this issues is really resolved, please close

**Comment by mbausch0** [comment]
Hi all,

thanks for your input.
I am sorry for the long delay in my responding to your input.

Just now, I had the chance to test that, still on Version 2.11:
Appending a trailing "#" to the syntax suggested by @daniceexi, I managed to have xcat generate a grub2 config file with all attributes specified as desired.

What still remains missing, is the autoyast file.
@daniceexi what do you mean with:
> You are using boottarget method, which means xCAT does not know which OS the you want to install.
> Then xCAT cannnot figure out how to generate the configuration file like autoyast for you.

I am doing an osimage deployment.
As far as I understand, my osimage definition does specify, which OS I want to install.
For details of my osimage definition (myimage), please see this issue's history.

Am I supposed to create the autoyast file myself?
Or is there any way to enable xcat doing this automatically?
Is that dealt with in the 2.11.2 milestone?



The details of what I did to successfully have xcat create a grub2 boot onfig file, are these:
```
ismgp92:~ # tabdump boottarget
#bprofile,kernel,initrd,kcmdline,comments,disable
"mytarget","xcat/osimage/myimage/linux64","xcat/osimage/myimage/initrd64","quiet sysrq=1 insmod=sym53c8xx insmod=ipr autoyast=http://!myipfn!:80/install/autoinst/#NODEATTRIB:nodelist:node# install=http://!myipfn!:80/install/repos/sles11sp4sap/ppc64/1 netdevice=#NODEATTRIB:mac:mac# dhcptimeout=150",,

ismgp92:~ # nodeset test880m osimage=myimage
test880m: install boottarget-ppc64-mytarget

ismgp92:~ # cat /tftpboot/boot/grub2/test880m
#install boottarget-ppc64-mytarget
set timeout=5
set default="xCAT OS Deployment"
menuentry "xCAT OS Deployment" {
    insmod http
    insmod tftp
    set root=tftp,10.250.176.198
    echo Loading Install kernel ...
    linux /xcat/osimage/myimage/linux64 quiet sysrq=1 insmod=sym53c8xx insmod=ipr autoyast=http://10.250.176.198:80/install/autoinst/test880m install=http://10.250.176.198:80/install/repos/sles11sp4sap/ppc64/1 netdevice=22:02:8E:6D:46:02 dhcptimeout=150
    echo Loading initial ramdisk ...
    initrd /xcat/osimage/myimage/initrd64
```

As mentioned above, the autoyast file was not created:
```
ismgp92:~ # ls -l /install/autoinst/test880m
ls: cannot access /install/autoinst/test880m: No such file or directory
```

**Comment by daniceexi** [comment]
@mbausch0 You need to specify the kernel parameter by yourself to specify the location of kickstart file. Then you need to generate the kickstart file by yourself too since xCAT does not know anything about your target OS. 

I am curious why you need to use boottarget instead of anaconda template (rh/centos).
If your system is using kickstart, you can try to copy one from a rh target node.

**Comment by mbausch0** [comment]
@daniceexi Good question :=)

I am using SUSE/autoyast, not RH/kickstart, but the idea of your question still applies.
Also, it is important to note, that I am using ppc64 (IBM Power8) hardware.

On this hardware, TFTP transfers during net-boot do have a size limit per transferred file.
IIRC, this limit is somewhere around 32MB per file.
I saw TFTP transfers of larger files fail for both, yaboot and grub2.
netbooting the kernel directly from firmware also doesn't succeed.

Now, the kernel which xcat uses by default to netboot SLES 11.4 is too large and thus netboot fails.

The solution is simple: do netboot the same way as the original SLES 11.4 .iso image does DVD boot: boot a different (smaller) kernel and add in an initramdisk. This way you will TFTP two files, but each of them small enough to succeed. Also, you are closer to what the original .iso image does.

To implement this solution in an xcat environment, my idea was also simple:
Utilize osimage and boottarget features to tell xcat to use a different kernel and an initramdisk.
Then do all the other install work the same way as I "always" used to do with SLES 11 on x86_64: have xcat generate an autoyast profile, execute pre/post scripts and so on.


During this github issue, it seems to turn out, that specifying custom kernel+initrd will stop xcat from generating the autoyast profile.
This is not, what I originally intended. I want to change boot kernel, initramdisk and kcmdline, not the whole deployment process (with which I am satisfied from my prior x86_64 experience).
If I generate the autoyast profile myself, then I will end up writing code to perform work, which xcat already is able to do well.

How could this be solved in a better way?
- Follow the same deployment process with autoyast profiles and all that stuff even when specifying custom boottarget attributes?
- Changing xcat defaults for netbooting SLES 11 on ppc64 so that I wouldn't need to specify custom kernel, initrd, kcmdline?
- Other ideas?

**Comment by daniceexi** [comment]
@mbausch0 xCAT known the problem you mentioned that yaboot has problem to download the file which is larger than 32M. So we involved grub2.
For sles11.4 ppc64, the grub2 should be used in the **netboot*8 attribute.  Also the grub2-http also can be tried. 

~~~~
                       Arch                  OS                           valid netboot options
                       x86, x86_64           ALL                          pxe, xnba
                       ppc64                 <=rhel6, <=sles11.3          yaboot
                       ppc64                 >=rhels7, >=sles11.4         grub2,grub2-http,grub2-tftp
~~~~

**Comment by mbausch0** [comment]
@daniceexi 
Did you successfully test netbooting SLES11.4 on a ppc64 node with grub2 or grub2-http?
On which xcat version?

In my previous tests, grub2 demonstrated to have the same problem downloading files larger than 32MB as yaboot does.
The same goes for grub2-http, which I tested today:
This also does not succeed in downloading the kernel, if it is too large.
It appears to do an incomplete download of the kernel file:
The web server's access_log shows **"HTTP 206 Partial Content"**.

Therefore, grub2-http does not appear to solve the 32MB per file problem.
Also, nodeset (at least in my version) does complain about grub2-http being invalid.
And I had to manually add the bootfile to DHCP because that was missing.

The only way of netbooting ppc64 that I have seen working so far, is to boot kernel+initrd where each of the files is smaller than 32MB.
But then: How to instruct xcat to do so without changing the rest of the deployment process?

To test grub2-http, I used the original osimage definition as created by copycds, not involving boottarget features:
```
ismgp92:~ # lsdef -v
lsdef - Version 2.11 (git commit 9ea36ca6163392bf9ab684830217f017193815be, built Mon Nov 30 05:43:11 EST 2015)

ismgp92:~ # lsdef test880m
Object name: test880m
    arch=ppc64
    groups=lpar
    mac=22:02:8E:6D:46:02
    netboot=grub2-http
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles

ismgp92:~ # nodeset test880m osimage=sles11sp4sap-ppc64-install-compute
Warning: test880m: grub2-http might be invalid when provisioning sles11sp4sap-ppc64-install-compute,valid options: "yaboot".
For more details see the 'netboot' description in the output of "tabdump -d noderes".
test880m: install sles11sp4sap-ppc64-compute

ismgp92:~ # cat /tftpboot/boot/grub2/test880m
#install sles11sp4sap-ppc64-compute
set timeout=5
set default="xCAT OS Deployment"
menuentry "xCAT OS Deployment" {
    insmod http
    insmod tftp
    set root=http,10.250.176.198
    echo Loading Install kernel ...
    linux /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 quiet autoyast=http://10.250.176.198:80/install/autoinst/test880m install=http://10.250.176.198:80/install/repos/sles11sp4sap/ppc64/1 netdevice=22:02:8E:6D:46:02  dhcptimeout=150
    echo Loading initial ramdisk ...
}

ismgp92:~ # ls -lh /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64
-rw-r--r-- 1 root root 34M Apr 28 14:41 /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64

ismgp92:~ # vi /var/lib/dhcp/db/dhcpd.leases
## manually add bootfile to the node's block:
host test880m {
  dynamic;
  hardware ethernet 22:02:8e:6d:46:02;
  fixed-address 10.250.176.195;
        supersede server.ddns-hostname = "test880m";
        supersede host-name = "test880m";
        supersede server.filename = "/boot/grub2/grub2-test880m";
}

ismgp92:~ # service dhcpd restart

## manually start netboot on the node

ismgp92:~ # tail -f /var/log/apache2/access_log
10.250.176.195 - - [28/Apr/2016:14:52:32 +0200] "GET /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 HTTP/1.1" 200 34720417 "-" "GRUB 2.02~beta2"
10.250.176.195 - - [28/Apr/2016:14:52:32 +0200] "GET /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 HTTP/1.1" 206 34654881 "-" "GRUB 2.02~beta2"
```


On the netbooted ppc64 node's console, the following happened:
```
0 > boot net

TFTP BOOT ---------------------------------------------------
Server IP.....................10.250.176.198
Client IP.....................10.250.176.195
Gateway IP....................10.250.176.1
Subnet Mask...................255.255.252.0
( 1  ) Filename................./boot/grub2/grub2-test880m
TFTP Retries..................5
Block Size....................512
FINAL PACKET COUNT = 312
FINAL FILE SIZE = 159544  BYTES

Elapsed time since release of system processors: 123812 mins 28 secs

      xCAT OS Deployment

      Use the ^ and v keys to change the selection.
      Press 'e' to edit the selected item, or 'c' for a command prompt.
   The selected entry will be started automatically in 0s.

Loading Install kernel ...
Loading initial ramdisk ...

SuSE Linux zImage starting: loaded at 086e0000-0a7f7d30 (0/0/00c3da70; sp: 069dfe10)
uncompressing ELF header done. (00000100 bytes)
Can't allocate memory for kernel image !

EXIT called ok
0 >
```

**Comment by daniceexi** [comment]
Please refer to the xcat bug: https://sourceforge.net/p/xcat/bugs/4420
 
It's possible that you need to update your firmware. What's the hardware type for server? P7 or P8? 

 
ThanksBest Regards----------------------------------------------------------------------Wang Xiaopeng (王晓朋)IBM China System Technology LaboratoryTel: 86-10-82453455Email: wxp@cn.ibm.comAddress: 28,ZhongGuanCun Software Park,No.8 Dong Bei Wang West Road, Haidian District Beijing P.R.China 100193
 
 
----- Original message -----From: mbausch0 <notifications@github.com>To: xcat2/xcat-core <xcat-core@noreply.github.com>Cc: Xiao Peng Wang/China/IBM@IBMCN, Mention <mention@noreply.github.com>Subject: Re: [xcat2/xcat-core] [Customer]failing to specify custom kernel, initrd, kcmdline via boottarget (#553)Date: Fri, Apr 29, 2016 12:37 AM 
@daniceexiDid you successfully test netbooting SLES11.4 on a ppc64 node with grub2 or grub2-http?On which xcat version?
In my previous tests, grub2 demonstrated to have the same problem downloading files larger than 32MB as yaboot does.The same goes for grub2-http, which I tested today:This also does not succeed in downloading the kernel, if it is too large.It appears to do an incomplete download of the kernel file:The web server's access_log shows "HTTP 206 Partial Content".
Therefore, grub2-http does not appear to solve the 32MB per file problem.Also, nodeset (at least in my version) does complain about grub2-http being invalid.And I had to manually add the bootfile to DHCP because that was missing.
The only way of netbooting ppc64 that I have seen working so far, is to boot kernel+initrd where each of the files is smaller than 32MB.But then: How to instruct xcat to do so without changing the rest of the deployment process?
To test grub2-http, I used the original osimage definition as created by copycds, not involving boottarget features:
ismgp92:~ # lsdef -vlsdef - Version 2.11 (git commit 9ea36ca6163392bf9ab684830217f017193815be, built Mon Nov 30 05:43:11 EST 2015)ismgp92:~ # lsdef test880mObject name: test880m    arch=ppc64    groups=lpar    mac=22:02:8E:6D:46:02    netboot=grub2-http    postbootscripts=otherpkgs    postscripts=syslog,remoteshell,syncfilesismgp92:~ # nodeset test880m osimage=sles11sp4sap-ppc64-install-computeWarning: test880m: grub2-http might be invalid when provisioning sles11sp4sap-ppc64-install-compute,valid options: "yaboot".For more details see the 'netboot' description in the output of "tabdump -d noderes".test880m: install sles11sp4sap-ppc64-computeismgp92:~ # cat /tftpboot/boot/grub2/test880m#install sles11sp4sap-ppc64-computeset timeout=5set default="xCAT OS Deployment"menuentry "xCAT OS Deployment" {    insmod http    insmod tftp    set root=http,10.250.176.198    echo Loading Install kernel ...    linux /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 quiet autoyast=http://10.250.176.198:80/install/autoinst/test880m install=http://10.250.176.198:80/install/repos/sles11sp4sap/ppc64/1 netdevice=22:02:8E:6D:46:02  dhcptimeout=150    echo Loading initial ramdisk ...}ismgp92:~ # ls -lh /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64-rw-r--r-- 1 root root 34M Apr 28 14:41 /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64ismgp92:~ # vi /var/lib/dhcp/db/dhcpd.leases## manually add bootfile to the node's block:host test880m {  dynamic;  hardware ethernet 22:02:8e:6d:46:02;  fixed-address 10.250.176.195;        supersede server.ddns-hostname = "test880m";        supersede host-name = "test880m";        supersede server.filename = "/boot/grub2/grub2-test880m";}ismgp92:~ # service dhcpd restart## manually start netboot on the nodeismgp92:~ # tail -f /var/log/apache2/access_log10.250.176.195 - - [28/Apr/2016:14:52:32 +0200] "GET /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 HTTP/1.1" 200 34720417 "-" "GRUB 2.02~beta2"10.250.176.195 - - [28/Apr/2016:14:52:32 +0200] "GET /tftpboot/xcat/osimage/sles11sp4sap-ppc64-install-compute/inst64 HTTP/1.1" 206 34654881 "-" "GRUB 2.02~beta2"
On the netbooted ppc64 node's console, the following happened:
0 > boot netTFTP BOOT ---------------------------------------------------Server IP.....................10.250.176.198Client IP.....................10.250.176.195Gateway IP....................10.250.176.1Subnet Mask...................255.255.252.0( 1  ) Filename................./boot/grub2/grub2-test880mTFTP Retries..................5Block Size....................512FINAL PACKET COUNT = 312FINAL FILE SIZE = 159544  BYTESElapsed time since release of system processors: 123812 mins 28 secs      xCAT OS Deployment      Use the ^ and v keys to change the selection.      Press 'e' to edit the selected item, or 'c' for a command prompt.   The selected entry will be started automatically in 0s.Loading Install kernel ...Loading initial ramdisk ...SuSE Linux zImage starting: loaded at 086e0000-0a7f7d30 (0/0/00c3da70; sp: 069dfe10)uncompressing ELF header done. (00000100 bytes)Can't allocate memory for kernel image !EXIT called ok0 >

**Comment by mbausch0** [comment]
We're using 9119-MHE (IBM Power8 E880) hardware.
We are at firmware level FW840.00 (056) from Dec 4, 2015,
which is more recent than xcat bug: https://sourceforge.net/p/xcat/bugs/4420

This bug refers to Nigel's blog post at
https://www.ibm.com/developerworks/community/blogs/aixpert/entry/suse_sles_11_on_power8_install_can_t_allocate_memory_for_kernel_image?lang=en
Nigel states (emphasize is from me):
> This is a known problem and it was installing on my POWER8 **but about the time we upgraded the machine to the latest System Firmware it stopped installing** from the regular media.

Alltogether, this doesn't give an indication to do a firmware update.

If you further follow Nigel's blog post, you will be advised to boot your node from SLES-11-SP3-MINI-ISO-power8.iso, downloaded from http://drivers.suse.com/ibm/IBM-POWER/sle11-sp3/install-readme.html

And there we are:
What this SLES-11-SP3-MINI-ISO-power8.iso does for SLES 11.3 is exactly the same than what the regular SLES 11.4 .iso does for 11.4 and also the same than what I am trying to do in xcat:
Boot a smaller kernel (linux64, not inst64) and add an initramdisk:
```

# download https://drivers.suse.com/ibm/IBM-POWER/sle11-sp3/SLES-11-SP3-MINI-ISO-power8.iso

ismgp92:~ # mount -ro loop /opt/iso/SLES-11-SP3-MINI-ISO-power8.iso /mnt

ismgp92:~ # cat /mnt/suseboot/yaboot.txt

  Welcome to SUSE Linux Enterprise 11 SP3!

  Type  "install"  to start the YaST installer on this CD/DVD
  Type  "slp"      to start the YaST install via network
  Type  "rescue"   to start the rescue system on this CD/DVD

ismgp92:~ # cat /mnt/suseboot/yaboot.cnf
message=yaboot.txt

image[64bit]=linux64
  initrd=initrd64
  label=install
  append="quiet sysrq=1 insmod=sym53c8xx insmod=ipr            "
image[64bit]=linux64
  initrd=initrd64
  label=slp
  append="quiet sysrq=1 install=slp           "
image[64bit]=linux64
  initrd=initrd64
  label=rescue
  append="quiet sysrq=1 rescue=1              "

ismgp92:~ # ls -lh /mnt/suseboot/initrd64 /mnt/suseboot/inst64 /mnt/suseboot/linux64
-r--r--r-- 1 root root 26M Jun  4  2014 /mnt/suseboot/initrd64
-r-xr-xr-x 1 root root 31M Jun  4  2014 /mnt/suseboot/inst64
-r--r--r-- 1 root root 20M Jun 14  2013 /mnt/suseboot/linux64
```

**Comment by daniceexi** [comment]
We did test the sles11.4. Did you try the sles11.4 on your server? If you still encounter problem, we can verify it again.
ThanksBest Regards----------------------------------------------------------------------Wang Xiaopeng (王晓朋)IBM China System Technology LaboratoryTel: 86-10-82453455Email: wxp@cn.ibm.comAddress: 28,ZhongGuanCun Software Park,No.8 Dong Bei Wang West Road, Haidian District Beijing P.R.China 100193
 
 
----- Original message -----From: mbausch0 <notifications@github.com>To: xcat2/xcat-core <xcat-core@noreply.github.com>Cc: Xiao Peng Wang/China/IBM@IBMCN, Mention <mention@noreply.github.com>Subject: Re: [xcat2/xcat-core] [Customer]failing to specify custom kernel, initrd, kcmdline via boottarget (#553)Date: Sat, Apr 30, 2016 1:08 AM 
We're using 9119-MHE (IBM Power8 E880) hardware.We are at firmware level FW840.00 (056) from Dec 4, 2015,which is more recent than xcat bug: https://sourceforge.net/p/xcat/bugs/4420
This bug refers to Nigel's blog post athttps://www.ibm.com/developerworks/community/blogs/aixpert/entry/suse_sles_11_on_power8_install_can_t_allocate_memory_for_kernel_image?lang=enNigel states (emphasize is from me):
This is a known problem and it was installing on my POWER8 but about the time we upgraded the machine to the latest System Firmware it stopped installing from the regular media.
Alltogether, this doesn't give an indication to do a firmware update.
If you further follow Nigel's blog post, you will be advised to boot your node from SLES-11-SP3-MINI-ISO-power8.iso, downloaded from http://drivers.suse.com/ibm/IBM-POWER/sle11-sp3/install-readme.html
And there we are:What this SLES-11-SP3-MINI-ISO-power8.iso does for SLES 11.3 is exactly the same than what the regular SLES 11.4 .iso does for 11.4 and also the same than what I am trying to do in xcat:Boot a smaller kernel (linux64, not inst64) and add an initramdisk:
# download https://drivers.suse.com/ibm/IBM-POWER/sle11-sp3/SLES-11-SP3-MINI-ISO-power8.isoismgp92:~ # mount -ro loop /opt/iso/SLES-11-SP3-MINI-ISO-power8.iso /mntismgp92:~ # cat /mnt/suseboot/yaboot.txt  Welcome to SUSE Linux Enterprise 11 SP3!  Type  "install"  to start the YaST installer on this CD/DVD  Type  "slp"      to start the YaST install via network  Type  "rescue"   to start the rescue system on this CD/DVDismgp92:~ # cat /mnt/suseboot/yaboot.cnfmessage=yaboot.txtimage[64bit]=linux64  initrd=initrd64  label=install  append="quiet sysrq=1 insmod=sym53c8xx insmod=ipr            "image[64bit]=linux64  initrd=initrd64  label=slp  append="quiet sysrq=1 install=slp           "image[64bit]=linux64  initrd=initrd64  label=rescue  append="quiet sysrq=1 rescue=1              "ismgp92:~ # ls -lh /mnt/suseboot/initrd64 /mnt/suseboot/inst64 /mnt/suseboot/linux64-r--r--r-- 1 root root 26M Jun  4  2014 /mnt/suseboot/initrd64-r-xr-xr-x 1 root root 31M Jun  4  2014 /mnt/suseboot/inst64-r--r--r-- 1 root root 20M Jun 14  2013 /mnt/suseboot/linux64

**Comment by mbausch0** [comment]
Yes, I tried sles 11.4.
sles11.4 is all, what I am talking about in this issue.
Booting sles 11.4 works, if I manually change yaboot or grub2 config to boot a smaller kernel plus an initramdisk. It doesn't work as long as xcat defaults of booting a lerger kernel are used.
I still don't manage to configure xcat such that it uses the working way of booting without breaking the rest of osimage deployment (generate autoyast porfile, ....)

The only reason why sles11.3 got involved in this discussion is that (AFAIK) this ppc64 filesize limit first occured on 11.3 ad was first solved on 11.3. But the solution is still the same in 11.4.

**Comment by daniceexi** [comment]
There was an xCAT bug for failed to install sles11.4 on P8 server. https://sourceforge.net/p/xcat/bugs/4420/
 
Use grub2, sles11.4 install for P7 works well.
 
xCAT team is trying to figure out a patch for this installer file size issue.
ThanksBest Regards----------------------------------------------------------------------Wang Xiaopeng (王晓朋)IBM China System Technology LaboratoryTel: 86-10-82453455Email: wxp@cn.ibm.comAddress: 28,ZhongGuanCun Software Park,No.8 Dong Bei Wang West Road, Haidian District Beijing P.R.China 100193
 
 
----- Original message -----From: mbausch0 <notifications@github.com>To: xcat2/xcat-core <xcat-core@noreply.github.com>Cc: Xiao Peng Wang/China/IBM@IBMCN, Mention <mention@noreply.github.com>Subject: Re: [xcat2/xcat-core] [Customer]failing to specify custom kernel, initrd, kcmdline via boottarget (#553)Date: Tue, May 3, 2016 5:38 PM 
Yes, I tried sles 11.4.sles11.4 is all, what I am talking about in this issue.Booting sles 11.4 works, if I manually change yaboot or grub2 config to boot a smaller kernel plus an initramdisk. It doesn't work as long as xcat defaults of booting a lerger kernel are used.I still don't manage to configure xcat such that it uses the working way of booting without breaking the rest of osimage deployment (generate autoyast porfile, ....)
The only reason why sles11.3 got involved in this discussion is that (AFAIK) this ppc64 filesize limit first occured on 11.3 ad was first solved on 11.3. But the solution is still the same in 11.4.

**Comment by xuweibj** [comment]
#1061 Solved this problem.
@junxiawang  has verified it, pass.
